src/fight/mod.rs

```diff
@@ -0,0 +1,1058 @@
+//! # Fight Module
+//!
+//! This module contain all the functions to emulate a fight between 2 Regiment
+//! This may evolve to multiple Regiment against multiple Regiment
+
+mod global_values;
+use crate::{model, regiment, roll};
+
+fn get_nb_attacks(regiment_attacking: &regiment::Regiment) -> usize {
+    regiment_attacking.get_model().get_stats().get_attack() * regiment_attacking.get_cols()
+        + ((regiment_attacking.get_model().get_stats().get_attack() * regiment_attacking.get_cols()
+            / 2) as usize)
+}
+
+fn get_nb_wounds(nb_attacks: usize, to_hit: usize, to_wound: usize) -> usize {
+    let nb_hit: usize = (((nb_attacks * (global_values::DEFAULT_DICE - to_hit + 1))
+        / global_values::DEFAULT_DICE) as f64)
+        .ceil() as usize;
+    (((nb_hit * (global_values::DEFAULT_DICE - to_wound + 1)) / global_values::DEFAULT_DICE) as f64)
+        .ceil() as usize
+}
+
+fn get_final_result(
+    nb_wounds: usize,
+    regiment_defending_armor: usize,
+    regiment_attacking_armor_penetration: usize,
+) -> usize {
+    let to_save: i8 = regiment_defending_armor as i8 - regiment_attacking_armor_penetration as i8;
+    let throw_to_save: i8 = match to_save {
+        i8::MIN..=0 => return nb_wounds,
+        1 => 6,
+        2 => 5,
+        3 => 4,
+        4 => 3,
+        5..=i8::MAX => 2,
+    };
+    let nb_save: usize = ((nb_wounds) * (global_values::DEFAULT_DICE - throw_to_save as usize + 1))
+        / global_values::DEFAULT_DICE;
+    nb_wounds - nb_save
+}
+
+fn fight_first_turn(
+    regiment_attacking: &regiment::Regiment,
+    regiment_defending: &regiment::Regiment,
+) -> usize {
+    let regiment_attacking_stats: &model::Stats = regiment_attacking.get_model().get_stats();
+    let regiment_defending_stats: &model::Stats = regiment_defending.get_model().get_stats();
+    let to_hit: usize = roll::compute_roll_to_hit(
+        regiment_attacking_stats.get_offensive(),
+        regiment_defending_stats.get_defense(),
+    );
+    let to_wound: usize = roll::compute_roll_to_wound(
+        regiment_attacking_stats.get_strength(),
+        regiment_defending_stats.get_resilience(),
+    );
+    let nb_attacks: usize = get_nb_attacks(regiment_attacking);
+    let nb_wounds: usize = get_nb_wounds(nb_attacks, to_hit, to_wound);
+    let final_result: usize = get_final_result(
+        nb_wounds,
+        regiment_defending_stats.get_armour(),
+        regiment_attacking_stats.get_armour_penetration(),
+    );
+    final_result
+}
+
+fn find_the_fastest(model_1: &model::Model, model_2: &model::Model) -> usize {
+    let model_1_agility: usize = model_1.get_stats().get_agility();
```

Comment:
Achtung, According to the rules, if the attacking unit is charging, it has a +1 bonus in speed
You can check but I think remember reading it

Reply:
True ! I'll change that in the code